Sources/JobsPostgres/PostgresJobsQueue.swift

```diff
@@ -302,6 +310,18 @@ extension PostgresJobQueue {
     public func makeAsyncIterator() -> AsyncIterator {
         return .init(queue: self)
     }
+
+    private func startTransaction(connection: PostgresConnection) async throws {
+        try await connection.query("BEGIN;", logger: self.logger)
+    }
+
+    private func commitTransaction(connection: PostgresConnection) async throws {
+        try await connection.query("COMMIT;", logger: self.logger)
+    }
+
+    private func rollbackTransaction(connection: PostgresConnection) async throws {
```

Comment:
Could we not combine these three functions into one function `PostgresClient.withTransaction(logger:)` eg
```swift
extension PostgresClient {
    func withTransaction<Value>(logger: Logger, _ process: (PostgresConnection) async throws -> Value) async throws -> Value {
        try await withConnection { connection in
            do {
                try await connection.query("BEGIN;", logger: logger)
                let value = try await process(connection)
                try await connection.query("COMMIT;", logger: logger)
                return value
            } catch {
                try await connection.query("ROLLBACK;", logger: logger)
                throw error
            }
        }
    }
}
```

Reply:
That's brilliant! I will update the PR.

Reply:
I think this util function should extracted and move to PostgresNIO.